### ISSUE TITLE
Training restarts and checkpoint file is corrupted

### ISSUE THREAD

**Issue by LoekvanKooten** [pytorch/fairseq#2684]
## 🐛 Bug

fairseq regularly but completely randomly fails on Asian datasets.

### To Reproduce
Train an Asian dataset (Japanese, Chinese, Korean).

Steps to reproduce the behavior (**always include the command you ran**):

1. Run cmd 'fairseq-train /home/loek/modernmt4/runtime/koreng/tmp/training/data_generated --save-dir /home/loek/modernmt4/runtime/koreng/tmp/training/_temp_train/nn_model --task mmt_translation --user-dir /home/loek/modernmt4/build/mmt-4.5.4.jar/mmt --share-all-embeddings --no-progress-bar --tensorboard-logdir /home/loek/modernmt4/runtime/koreng/tmp/training/_temp_train/tensorboard_logdir --dataset-impl mmap --arch transformer_mmt_base --clip-norm 0.0 --label-smoothing 0.1 --attention-dropout 0.1 --dropout 0.3 --weight-decay 0.0 --criterion label_smoothed_cross_entropy --optimizer adam --adam-betas (0.9, 0.98) --log-interval 100 --lr 0.0005 --lr-scheduler inverse_sqrt --min-lr 1e-09 --warmup-init-lr 1e-07 --warmup-updates 4000 --max-tokens 3072 --update-freq 4 --save-interval-updates 1000 --keep-interval-updates 10 --keep-last-epochs 10'
2. See error

See attached training.log
[training.log](https://github.com/pytorch/fairseq/files/5316669/training.log)

#### Code sample
Fairseq is direct controlled from ModernMT, the code of which is freely available on GitHub.

### Expected behavior
Training should complete succesfully.

### Environment

 - fairseq Version (e.g., 1.0 or master):
As included by newest version of ModernMT. (ModernMT says this issue is caused by fairseq).
 - PyTorch Version (e.g., 1.0)
As included by newest version of ModernMT. (ModernMT says this issue is caused by fairseq).
 - OS (e.g., Linux):
Ubuntu 18.04.5 LTS
 - How you installed fairseq (`pip`, source):
As included by newest version of ModernMT. (ModernMT says this issue is caused by fairseq).
 - Build command you used (if compiling from source):
As included by newest version of ModernMT. (ModernMT says this issue is caused by fairseq).
 - Python version:
3.6
 - CUDA/cuDNN version:
10.1/v7.6.5
 - GPU models and configuration:
Gigabyte GeForce AORUS RTX 2080 Ti XTREME 11G

For more info, also see this thread on https://github.com/modernmt/modernmt/issues/562, including information about what the developers of ModernMT and I tried to solve this issue.

**Comment by myleott** [comment]
Looking at the log, it seems to have trained for 71 epochs, and then I just see `^@^@^@^@^@^@^@^@^@^@^@^@^@^@^` and the job restarts. After the restart, it tries to load the checkpoint but it seems the file is corrupted.

It's not clear what's going on, but can you try loading the checkpoint manually and see if it works:

```bash
$ python
>>> import torch
>>> cpt = torch.load('/home/loek/modernmt4/runtime/koreng/tmp/training/_temp_train/nn_model/checkpoint_63_1000.pt')

**Comment by LoekvanKooten** [comment]
The same error kept repeating. I then deleted everything and the same issue happened. I then deleted everything again and then it worked.

This must have to do something with the training going differently every time, and some combinations of random values making fairseq crash. Or maybe it's the timing between the CPU/GPU?
